tests/50federation/36state.pl

```diff
@@ -835,45 +822,31 @@ sub get_state_ids_from_server {
             )->then( sub {
                my ( $req ) = @_;
 
+               my @state = values( %{ $room->{current_state} } );
                my $resp = {
                   pdu_ids => [
-                     map { $_->{event_id} } values( %{ $room->{current_state} } ),
+                     map { $_->{event_id} } @state,
                   ],
                   auth_chain_ids => [],
                };
 
                log_if_fail "/state_ids response", $resp;
 
+               # once we respond to `/state_ids`, the server may send a /state request;
+               # be prepared to answer that.  (it may, alternatively, send individual
+               # /event requests)
+               $state_req_fut = await_and_handle_request_state(
+                  $inbound_server, $room, $missing_event_y->{event_id}, \@state
+               );
```

Comment:
Does sending the real auth chain here defeat the purpose of the test?

Reply:
errm I don't *think* so, because synapse doesn't actually take a whole lot of notice of the auth_events in the response.

That said, I'll override the auth events here.